Services/UI/OJS.Services.Ui.Data/Implementations/SubmissionsDataService.cs

```diff
@@ -114,6 +114,12 @@ public int GetUserSubmissionTimeLimit(int participantId, int limitBetweenSubmiss
     public bool HasUserNotProcessedSubmissionForProblem(int problemId, string userId) =>
         this.DbSet.Any(s => s.ProblemId == problemId && s.Participant!.UserId == userId && !s.Processed);
 
+    public async Task<TServiceModel> GetProblemBySubmission<TServiceModel>(int submissionId)
+        => await this.GetByIdQuery(submissionId)
+            .Select(p => p.Problem)
+            .FirstOrDefaultAsync()
+            .Map<TServiceModel>();
```

Comment:
Map before materializing the data
Should be 
```
await this.GetByIdQuery(submissionId)
            .Select(p => p.Problem)
            .Map<TServiceModel>()
            .FirstOrDefaultAsync();
```

Reply:
Already tried that.. We cannot materialize after .Map.

Reply:
Why